db/migrate/20170629151616_add_unique_index_to_repos.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToRepos < ActiveRecord::Migration[5.1]
+  def change
+    add_index :repos, [:name, :user_name], :unique => true
```

Comment:
I'm already lowercasing name and user_name in a pre-commit hook in Ruby. That part of the validation was redundant.